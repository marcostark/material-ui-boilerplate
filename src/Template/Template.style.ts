import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
`

export const Content = styled.main`
  flex-grow: 1;
  padding: ${({ theme }) => theme.spacing(3)}px;
`
