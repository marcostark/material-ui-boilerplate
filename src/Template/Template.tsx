import React, { useState } from 'react'
import Header from 'components/Header'
import Footer from 'components/Footer'
import Routes from 'routes'
import * as S from './Template.style'

const Template = () => {
  const [isOpen, setIsOpen] = useState(false)

  const handleToggleDrawer = (openDefault: boolean) => (
    event: React.KeyboardEvent | React.MouseEvent
  ) => {
    if (
      event.type === 'keydown' &&
      ((event as React.KeyboardEvent).key === 'Tab' ||
        (event as React.KeyboardEvent).key === 'Shift')
    ) {
      return
    }

    setIsOpen(openDefault)
  }

  return (
    <S.Wrapper>
      <Header
        title="Home"
        isOpen={isOpen}
        onToggleDrawer={handleToggleDrawer}
      />
      <S.Content>
        <Routes />
      </S.Content>
      <Footer />
    </S.Wrapper>
  )
}

export default Template
