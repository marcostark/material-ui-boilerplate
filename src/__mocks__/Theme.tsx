import React, { ReactElement } from 'react'
import { StylesProvider, StylesOptions } from '@material-ui/styles'
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

const generateClassName: StylesOptions['generateClassName'] = (rule, sheet) =>
  `${sheet?.options.classNamePrefix ?? 'classname'}-${rule.key}`

type Props = {
  children: ReactElement | ReactElement[]
}

const Theme = ({ children }: Props) => (
  <StylesProvider generateClassName={generateClassName}>
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </MuiThemeProvider>
  </StylesProvider>
)

export default Theme
