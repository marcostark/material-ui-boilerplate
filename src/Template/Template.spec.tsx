import React from 'react'
import { Router } from 'react-router-dom'
import { createMemoryHistory, History, State } from 'history'
import { render } from '@testing-library/react'
import Theme from '__mocks__/Theme'
import Template from './Template'

describe('<Template />', () => {
  let history: History<State>

  beforeEach(() => {
    history = createMemoryHistory()
  })

  it('should render correctly', () => {
    const { container } = render(
      <Router history={history}>
        <Theme>
          <Template />
        </Theme>
      </Router>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
