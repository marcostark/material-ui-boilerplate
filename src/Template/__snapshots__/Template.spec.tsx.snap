// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Template /> should render correctly 1`] = `
.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  margin-right: 16px;
}

.c2 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: none;
}

.c6 {
  width: 100%;
  margin-left: 0;
  margin-right: 16px;
  position: relative;
  background-color: rgba(255,255,255,0.15);
  border-radius: 4px;
}

.c6:hover {
  background-color: rgba(255,255,255,0.25);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  padding: 0px 16px;
  position: absolute;
  pointer-events: none;
}

.c8 {
  display: none;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  padding: 16px;
  text-align: center;
}

.c11 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 24px;
}

@media (min-width:600px) {
  .c5 {
    display: block;
  }
}

@media (min-width:600px) {
  .c6 {
    width: auto;
    margin-left: 24px;
  }
}

@media (min-width:960px) {
  .c8 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:960px) {
  .c9 {
    display: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular c2 MuiToolbar-gutters"
      >
        <div
          class="c3"
        >
          <button
            aria-label="Open drawer"
            class="MuiButtonBase-root MuiIconButton-root c4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root c5 MuiTypography-h6 MuiTypography-noWrap"
          >
            Home
          </h6>
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root makeStyles-inputRoot"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input makeStyles-inputInput"
              placeholder="Search…"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="c8"
        >
          <button
            aria-label="show 4 new mails"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <span
                class="MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
                >
                  4
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="show 17 new notifications"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <span
                class="MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
                >
                  17
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="primary-search-account-menu"
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="c9"
        >
          <button
            aria-controls="primary-search-account-menu-mobile"
            aria-haspopup="true"
            aria-label="show more"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
  <main
    class="c10"
  >
    <div
      class="c11"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </main>
  <footer
    class="c12"
  >
    © 2020
  </footer>
</div>
`
